docarray/array/mixins/group.py

```diff
@@ -64,3 +64,57 @@ def batch(
 
         for i in range(n_batches):
             yield self[ix[i * batch_size : (i + 1) * batch_size]]
+
+    def batch_indices(
+        self,
+        batch_size: int,
+        shuffle: bool = False,
+    ) -> Generator[list, None, None]:
```

Comment:
It's not, it can be removed or kept as an alternative to`batch_ids` since docarray suports both indexing by ids and indexing by indices. Probably for in_memory it's faster to do it by index and in sqlite to do it by ids.

Reply:
removed, we can restore it later if needed